bindings/java/src/main/com/apple/foundationdb/tuple/Tuple.java

```diff
@@ -260,8 +285,8 @@ public Tuple add(byte[] b, int offset, int length) {
 	 *
 	 * @return a newly created {@code Tuple}
 	 */
-	public Tuple addAll(List<? extends Object> o) {
-		List<Object> merged = new ArrayList<Object>(o.size() + this.elements.size());
+	public Tuple addAll(List<?> o) {
+		List<Object> merged = new ArrayList<>(o.size() + this.elements.size());
```

Comment:
I am curious why did you remove the type parameter here?

Reply:
On the second line, it's unnecessary. Starting in Java 7, the JDK will automatically detect the type parameter in instances like `A<B> v = new A<>()` and turn it into `A<B> v = new A<B>()` for you. So the two are equivalent, but the second is former is more idiomatic (starting with Java 7).

In the first line, as all types extend `Object`, a `List<? extends Object>` is the same type as `List<?>`, but the second is pithier.

Reply:
Java's syntax is becoming... weird. If they were to make it simple, why not just eliminate the required `<>` part too. So it looks like `List<Object> merged = new ArrayList(...)`

Reply:
Yeah, this Stack Overflow post says it has something to do with wanting to distinguish between raw types and generic types, but I'm not sure I'm convinced: https://stackoverflow.com/questions/4166966/what-is-the-point-of-the-diamond-operator-in-java-7